### PR TITLE
Feature/audio mix

### PR DESCRIPTION
**Pull request by rkersche**

### CODE REVIEW

File: bitmovin/resources/models/filters/audio_mix_channel.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    def source_channels(self):
+        return self._source_channels
+
+    @source_channels.setter
+    def source_channels(self, new_value):
+        if new_value is None:
+            return
+
+        if not isinstance(new_value, list):
+            raise InvalidTypeError('partitions has to be a list of AudioMixSourceChannel enums')
```
Comment (dominic-miglar) [inline bitmovin/resources/models/filters/audio_mix_channel.py]: partitions?

File: bitmovin/resources/models/filters/audio_mix_channel.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            source_channels = []
+            for item in new_value:
+                source_channels.append(item)
+            self._source_channels = source_channels
+        else:
+            self._source_channels = new_value
+
+    def serialize(self):
+        serialized = super().serialize()
+        serialized['channelNumber'] = self.channel_number
```
Comment (dominic-miglar) [inline bitmovin/resources/models/filters/audio_mix_channel.py]: is that actually needed?

File: bitmovin/resources/models/filters/audio_mix_filter.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+from . import AbstractFilter
+from bitmovin.utils import Serializable
+from bitmovin.errors import InvalidTypeError
+from bitmovin.resources.enums import AudioMixFilterChannelLayout
+from . import AudioMixChannel
```
Comment (dominic-miglar) [inline bitmovin/resources/models/filters/audio_mix_filter.py]: i would suggest to reorder this import statements

File: bitmovin/resources/models/filters/audio_mix_filter.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+
+class AudioMixFilter(AbstractFilter, Serializable):
+    def __init__(self, name=None, channel_layout=None, audio_mix_channels=None, id_=None,
```
Comment (dominic-miglar) [inline bitmovin/resources/models/filters/audio_mix_filter.py]: there's nothing required?

File: bitmovin/resources/models/filters/audio_mix_source_channel.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
+from . import AbstractFilter
```
Comment (dominic-miglar) [inline bitmovin/resources/models/filters/audio_mix_source_channel.py]: please put that relative imports to the bottom of the import section

File: bitmovin/resources/models/filters/audio_mix_source_channel.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        serialized = super().serialize()
+        serialized['channelNumber'] = self.channel_number
+        serialized['gain'] = self.gain
+        serialized['type'] = self.channel_type
+        return serialized
+
+    @classmethod
+    def parse_from_json_object(cls, json_object):
+        channel_type = json_object.get('type')
+        channel_number = json_object.get('channelNumber')
```
Comment (dominic-miglar) [inline bitmovin/resources/models/filters/audio_mix_source_channel.py]: is that needed here?

File: bitmovin/resources/models/filters/audio_mix_source_channel.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        gain = json_object.get('gain')
```
Comment (dominic-miglar) [inline bitmovin/resources/models/filters/audio_mix_source_channel.py]: is that needed here?

File: bitmovin/resources/models/filters/audio_mix_channel.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
+from bitmovin.errors import InvalidTypeError
+from bitmovin.utils import Serializable
+
+
+class AudioMixChannel(Serializable):
+    def __init__(self, channel_number=None, source_channels=None):
```
Comment (dominic-miglar) [inline bitmovin/resources/models/filters/audio_mix_channel.py]: i guess there are required attributes?

File: bitmovin/resources/models/filters/audio_mix_source_channel.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
+from . import AbstractFilter
+from bitmovin.utils import Serializable
+from bitmovin.errors import InvalidTypeError
+from bitmovin.resources.enums import AudioMixFilterChannelType
+
+
+class AudioMixSourceChannel(Serializable):
+    def __init__(self, channel_type=None, channel_number=None, gain=1.0):
```
Comment (dominic-miglar) [inline bitmovin/resources/models/filters/audio_mix_source_channel.py]: i guess there are required attributes?

File: bitmovin/resources/models/filters/audio_mix_channel.py (hunk 2 above)
Comment (rkersche) [inline bitmovin/resources/models/filters/audio_mix_channel.py]: Didn't work without that

File: bitmovin/resources/models/filters/audio_mix_source_channel.py (hunk 7 above)
Comment (rkersche) [inline bitmovin/resources/models/filters/audio_mix_source_channel.py]: Didn't work without that

File: bitmovin/resources/models/filters/audio_mix_source_channel.py (hunk 6 above)
Comment (rkersche) [inline bitmovin/resources/models/filters/audio_mix_source_channel.py]: Didn't work without that

**Pull request merged**